src/text-field-components.tsx

```diff
@@ -191,7 +191,8 @@ const useFieldContainerStyles = createUseStyles((theme) => ({
         },
         display: 'flex',
         position: 'relative',
-        backgroundColor: theme.colors.backgroundContainer,
+        backgroundColor: ({readOnly}) =>
+            readOnly ? theme.colors.backgroundAlternative : theme.colors.backgroundContainer,
```

Comment:
I think these are the same color in dark mode

Reply:
Yep, @aweell is reviewing this with the new inverse/alternative logic. Maybe in dark mode we have change boxed logic to keep border in this elements. But we are working on it.

Another solution to this is that readOnly appear with background color in dark mode to differentiate from default state
![image](https://user-images.githubusercontent.com/6722153/188957254-65a7c659-a792-4b22-bc99-9bdb58179ad2.png)

Reply:
we are going to use neutralLow for readOnly by the moment, this color in dark mode is slightly different than `backgroundAlternative`, which is not the best solution but we have to improve a lot dark mode colors (in general)

The usage of readOnly prop is not very common and by the moment is a commitment that we are willing to have in dark mode

![image](https://user-images.githubusercontent.com/6722153/191940935-3c67be36-890e-4f24-bd47-327915bac344.png)